### DISCUSSION TITLE
Side-by-side Translations

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3011] [Feature Requests]
When you have content translations, it is common practice to have a reference language that is used for the main content to translate. Currently, our translation interface opens in a modal/drawer, and so you can only see one language at a time, this discussion is about adding a way to set a "reference" language, that then can be shown "side-by-side", allowing a better translation flow.

UX Options:
* Show reference language values in a _notice_ below each translation field
    * Needs to handle longer values (force height, and scroll value)
* Use a tabbed approach on entire drawer for both languages (can't see both at same time)
* Use a tabbed approach on field for both languages (can't see both at same time)
* Force drawer edit form into **single column** to fit two side-by-side forms (semi standard)
* Open a wider drawer to fit two side-by-side forms (non standard)
* Keep editing experience on page, with two side-by-side forms (non standard)

**Comment by HCl-not-HCi** [comment]
I played around with the translations and realized that the current state is a bit cumbersome. Side-by-side translations would be really helpful.

> Keep editing experience on page, with two side-by-side forms (non standard)

I like this approach of showing and editing the translations on the original page side-by-side the most. But I'm not sure how easy it is to implement.

> Show reference language values in a notice below each translation field 

Maybe it would be easier to combine the first approach I mentioned together with this approach. You only show all translation values side-by-side but do not allow editing their values. For the actual editing you open a drawer where the reference language values are shown below each field (or side-by-side in a larger drawer).

I think a view where you can see all translations side-by-side (if you have enough space on your screen) is really helpful to keep an overview about your content.

**Comment by MoltenCoffee** [comment]
Quite a difficult conundrum. As @HCl-not-HCi stated as well, the current UX is not optimal and makes translating quite difficult. Besides that, the current UI seems really made to support pages with a primary language and secondary languages, instead of providing easy usage for websites that offer content in equal importance.

For example, a project I'm working on has articles in both English and Dutch. Most of them are written in Dutch, and a large portion is translated to English as well. However, it is likely at some point we'll have articles that are only in English (at least until they are translated) as well. If we create these in Directus, this would mean having an empty item with only the English translation filled in.

That's a shame, really, as right now, from a editor's point of view, the easiest way would be to just add multiple fields "title", "title_de", "title_nl" etc, which is a waste of the build in options. I personally find the current drawer/modal quite un-intuitive as well.

Personally, I think both a tabbed approach in the main view and side-by-side views are viable options, perhaps depending on screen size. 
If a drawer stays, perhaps a simple button to automatically add the original content to the translation fields would be a good option, as this would enable people to translate the text on the fly.

**Comment by benhaynes** [comment]
Just to follow up here, these are the current designs I'm working on. The idea is to allow for single or side-by-side translations (Interface Option could also define default reference language). This same GUI could be moved to the parent form (instead of the drawer), but that needs a bit more design work to clearly define which fields are "translated" vs agnostic.

![Item Detail](https://user-images.githubusercontent.com/522079/110666842-ac8e8c00-8197-11eb-8def-436ee55d4e7e.png)

![Item Detail2](https://user-images.githubusercontent.com/522079/110666859-b0221300-8197-11eb-9975-c12b059cfc67.png)

**Reply by MoltenCoffee** [reply]
Looking good @benhaynes !

One thing I still struggle to imagine, is what if there are a few languange agnostic fields like status and the like, and all text-content is in translations; I believe at the moment the collection overview would show only a summary of the agnostic fields, right? Or is there a way to change how that is displayed?

Edit:

It's not entirely on topic here, but for example in this case:
![image](https://user-images.githubusercontent.com/13321277/110759361-d0aca480-824d-11eb-931a-93df6e1fbcb4.png)

I think it would be awesome if the translations could show the languages themselves instead of the ID of the translation, and if there was the possibility of changing the display to allow showing one or two of the translated fields. (Or am I missing something?)

**Reply by HCl-not-HCi** [reply]
@MoltenCoffee there's a PR which solves the stuff you mentioned (I think): https://github.com/directus/directus/pull/3592

@benhaynes Looks nice :)

**Reply by benhaynes** [reply]
Exactly what @HCl-not-HCi said!

**Reply by benhaynes** [reply]
Just a note for later: I would like to have translations added (like the M2A), instead of all languages being shown before being created. This may not be an issue depending on our new design though.

**Comment by bcartier** [comment]
Another idea might be to add **Translations** section to the right-hand panel, allowing the user to view one or more languages at a time depending on their current task. If I'm translating, I could select the base language and one additional language at a time. Or if I'm only interested in changing language-agnostic fields like a dropdown, I can work in the base language only - keeping the form nice and short. I think this would work particularly well for 3+ languages, since you can deal with them individually or compare them to any other language as needed. 

![directus-translations](https://user-images.githubusercontent.com/48411/117380455-b5ee5900-aea7-11eb-8791-d009e19e2d67.jpg)

I'd suggest showing the Label for the field (Title, or Description for example) then the fields help text (if any), followed by a copy of the Interface for each language. You could condense the whitespace to keep them visually grouped as a single field.

**Reply by benhaynes** [reply]
I like it — especially because this maintains the field widths (instead of overriding them to half width), and because it allows for more than 2 languages on screen at once!

Not sure why someone would want multiple translations per collection, but this direction wouldn't work if they did, since they might have different language sets. However this could be overcome by having the "language selection" on the page instead of in the right info sidebar.

**Reply by MoltenCoffee** [reply]
I think this is beautiful @bcartier ! Very intuitive!

**Reply by rijkvanzanten** [reply]
@benhaynes We _could_ make `translations` a more system thing (instead of a regular m2m) to start limiting it to 1 per collection, and doing things like this a little easier 🤔 

I can't really think of a 80/20 use case where you'd have 2+ different translations setups per collection

**Reply by benhaynes** [reply]
Agree, no good use-case that I can think of. I think making translations more "system" is probably a good idea.
